src/test/ui/associated-type-bounds/auxiliary/fn-dyn-aux.rs

```diff
@@ -0,0 +1,182 @@
+// Traits:
```

Comment:
We should feature gate dyn types separately.

Reply:
Hmm, what's your reasoning? (Thoughts, @centril, since you wrote the tests and RFC?)

Reply:
Some context... Me, @nikomatsakis, and @tmandry had a Zoom call today where we went through the PR.

The thinking here is that `dyn Foo<Assoc: Bar>` introduces a type variable `X` and then constrains (`X: Bar`) it *unlike* the other cases where you induce a constraint `<T as Foo>::Assoc: Bar`. 

Because of this, @nikomatsakis thought that we might want to split the feature gates so that we can possibly decide and stabilize separately. I think that's pretty reasonable. If we decide in the end that we want to stabilize both together, we still can.

So if you can move `dyn Foo<Assoc: Bar>` to `associated_type_bounds_dyn` that would be neat.

Reply:
Okay. I'll try to do that when I get some time.